fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/runners/throttled/ThrottledRetryingIterator.java

```diff
@@ -444,11 +441,17 @@ private Builder(TransactionalRunner runner, Executor executor, ScheduledExecutor
         }
 
         private Builder(FDBDatabase database, FDBRecordContextConfig.Builder contextConfigBuilder, CursorFactory<T> cursorCreator, ItemHandler<T> singleItemHandler) {
-            this(new TransactionalRunner(database, contextConfigBuilder),
-                    database.newContextExecutor(contextConfigBuilder.getMdcContext()),
-                    database.getScheduledExecutor(),
-                    cursorCreator,
-                    singleItemHandler);
+            // Mandatory fields are set in the constructor. Everything else is optional.
+            this.database = database;
+            this.contextConfigBuilder = contextConfigBuilder;
+            this.cursorCreator = cursorCreator;
+            this.singleItemHandler = singleItemHandler;
+            // set defaults
```

Comment:
I suggest below that you remove the other constructor, but if not, perhaps worth settings these ones in the field declaration, and remove the duplication.